qiskit/transpiler/passes/remove_diagonal_gates_before_measurere.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2019, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+
+"""
+Transpiler pass to remove diagonal gates (like RZ, T, Z, etc) before
+a measurement. Including diagonal control gates.
+"""
+
+from qiskit.circuit import Measure
+from qiskit.extensions.standard import RZGate, ZGate, TGate, SGate, TdgGate, SdgGate, U1Gate,\
+    CzGate, CrzGate, Cu1Gate, RZZGate
+from qiskit.transpiler.basepasses import TransformationPass
+
+
+class RemoveDiagonalGatesBeforeMeasure(TransformationPass):
+    """Remove diagonal gates (like RZ, T, Z, etc) before a measurement.
+    Including diagonal control gates."""
+
+    def run(self, dag):
+        """Return a new circuit that has been optimized."""
+        diagonal_gates = (RZGate, ZGate, TGate, SGate, TdgGate, SdgGate, U1Gate)
+        diagonal_control_gates = (CzGate, CrzGate, Cu1Gate, RZZGate)
```

Comment:
non-blocking, but it's better to call these `one_qubit_diagonals` and `two_qubit_diagonals`. Rzz is not "controlled". And in general this works for any diagonal n-qubit gate followed by n measurements.

Reply:
Makes sense. Done in e63d4b2d